uint8 STOPPED = 0
uint8 RUNNING = 1
uint8 STARTING = 2
uint8 STOPPING = 3

uint8 state
time last_state_change
